openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -239,6 +264,63 @@ def export(self, request, **_kwargs) -> HttpResponse:
 
         return HttpResponse(tags, content_type=content_type)
 
+    @action(detail=False, url_path="import", methods=["post"])
+    def create_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Create a new taxonomy and imports the tags from the uploaded file.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
+
+        body = TaxonomyImportNewBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        taxonomy_name = body.validated_data["taxonomy_name"]
+        taxonomy_description = body.validated_data["taxonomy_description"]
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        taxonomy = create_taxonomy(taxonomy_name, taxonomy_description)
+        try:
+            import_success = import_tags(taxonomy, file, parser_format)
+
+            if import_success:
+                return HttpResponse(status=200)
+            else:
+                import_error = get_last_import_log(taxonomy)
+                taxonomy.delete()
+                return HttpResponseBadRequest(import_error)
+        except ValueError as e:
+            return HttpResponseBadRequest(e)
+
+    @action(detail=True, url_path="tags/import", methods=["put"])
+    def update_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Imports tags from the uploaded file to an already created taxonomy.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
+
+        body = TaxonomyImportBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        taxonomy = self.get_object()
+        try:
+            import_success = import_tags(taxonomy, file, parser_format)
+
+            if import_success:
+                return HttpResponse(status=200)
```

Comment:
Ditto here, can we return the Taxonomy on success?

Reply:
Yes, but I think we will need to revisit this in the future. 

I think we should return the `Taxonomy` for now, but we will probably want to return the tag list for this taxonomy (at least the first page of it) after the import.

We will have a cleaner view when implementing the Import inside the Taxonomy Details.

And just to make sure: the import inside the Taxonomy Details is not part of this task, right?

Reply:
@rpenido 

> We will have a cleaner view when implementing the Import inside the Taxonomy Details.

Agreed -- ok to just return Taxonomy for now.

> And just to make sure: the import inside the Taxonomy Details is not part of this task, right?

Correct, importing tags to an existing taxonomy is covered by https://github.com/openedx/modular-learning/issues/140